CONTRIBUTING.md

```diff
@@ -82,3 +90,55 @@ export class MyConstruct extends Construct {
   * Other import methods normally require access to the context API which is beyond the scope of these constructs. We'd like to address this in the future.
 * Constructs should implement an interface so that the import methods (e.g. `.fromSomethingAttributes()`) can return the interface type.
 
+
+
+## Testing
+
+Tests are fundamental to building good constructs. The following testing guidelines should be followed:
+
+* All tests should be stored in the `test/aws-<something>` subdirectory. For example, if you are creating
+a construct for the AppStream service, then the module name would be `aws-appstream` and tests would go in the `/test/aws-appstream` directory.
+* Fine-grained tests should be implemented for all constructs with a goal of 100% coverage. For example,
+if you were creating an Application construct in the AppSteam module, your tests should be in the `test/aws-appstream/application.test.ts` file.
+* Integration tests should be written for all constructs. For example, if you are creating an integration test
+for the Application construct in the AppStream module, your tests should be in the `test/aws-appstream/application.integ.ts` file.
+
+### Fine-grained assertion tests
+
+Unit tests should be created that cover all pathways of code in your construct, including the testing of any 
+default value or conditionals. Test reports will report code coverage. 
+
+For example, if your construct has a property that can be overridden, you should have a test that ensures the
+property is overridden correctly and that creates a default value when not overridden. 
+An example of this can be found [here](./test/aws-examplemodule/example.test.ts). 
+
+* In the test 'Uses provided Role for Lambda Function' the Lambda Function is referenced via the `.node.defaultChild` property and then the role is compared to the one provided.
+* In the test 'Creates a role when none provided' fine-grained assertions are used to verify a role was created with the correct 'AssumeRolePolicyDocument' property.
+* A `describe` is used to group together similar tests.
+
+All tests for a construct should be contained in a single file.
```

Comment:
If a construct uses helper functions, do we then want to have tests for the helper functions as well? If so, should they be in the same single file as the other tests?

Reply:
I believe they should. Anytime someone wants to know how XXX construct works they should be able to go to one place to find that, and that'd be the test file.

Reply:
This is one of the decisions I might challenge in the future, as the "easy to find" should already be solved by the folder structure we have. But I'm both fine to try it like this and to do the work of splitting tests into multiple files in case we decide for a different path in the future :)